correctme/ui/gui.py

```diff
@@ -1,146 +1,78 @@
-
-                    if len(self.prev2) >= 2:
-                        self.but_4.text = self.prev2[1]
-                    elif len(self.prev1) >= 2:
-                        self.but_4.text = self.prev2[1]
-
-                else:
-                    self.but_1.text = self.prev2[0]
-                    self.but_2.text = self.prev2[1]
-                    self.but_3.text = self.prev2[2]
-                    self.but_4.text = self.prev1[0]
-
-            print("BK tree suggestions\n")
-            print(self.prev1)
-            print("\n")
-
-            print("Double Metaphone suggestions\n")
-            print(self.prev2)
-            print("\n")
-
-            print("Intersection of both\n")
-            print(self.prev3)
-            print("---------------------------------------------------------------------------------------------------------------------------------------\n")
-
+    def on_release(self, text):
+        words = self.input_text.text.strip().split()
+        if len(words) > 0:
+            words[-1] = text
         else:
-            print("BK tree suggestions\n")
-            print(self.prev1)
-            print("\n")
-
-            print("Double Metaphone suggestions\n")
-            print(self.prev2)
-            print("\n")
-
-            print("Intersection of both\n")
-            print(self.prev3)
-            print("---------------------------------------------------------------------------------------------------------------------------------------\n")
-
-
-class SimpleKivy(App):
-    def __init__(self, dataset):
-        super(SimpleKivy, self).__init__()
+            words.append(text)
+        
+        self.input_text.text = ' '.join(words).strip()
+        self.input_text.focus = True
+
+    def on_text(self, text):
+        words = text.strip().split(' ')
+        last_word = words[-1]
+    
+        bktree_words, metaphone_words, intersect_words = sm.getSuggestion(last_word, self.tree, self.meta_dict)
+
+        # Need to form a list of four top words.
+        # TODO: Need to change this algo in future for n-grams.
+        i = 0
+        for word in intersect_words:
+            if i >= 4:
+                break
+            self.buttons[i].text = word 
+            i += 1
+        
+        for word in bktree_words:
+            if i >= 4:
+                break
+            self.buttons[i].text = word
+            i += 1
+
+        for word in metaphone_words:
+            if i >= 4:
+                break
+            self.buttons[i].text = word
+        
+        print("BK tree suggestions")
+        print(bktree_words)
+        print('\n')
+
+        print("Double Metaphone suggestions")
+        print(metaphone_words)
+        print('\n')
+
+        print("Intersection of both")
+        print(intersect_words)
+        print('\n')
+
+        print('-' * 50)
+
+
+class CorrectMeApp(App):
+    def __init__(self, dataset, **kwargs):
+        super(CorrectMeApp, self).__init__(**kwargs)
```

Comment:
I don't think it's necessary to pass arguments to `super` since Python3.0.

Reply:
Yes, you are right! Just read about it. Thanks for pointing that out.

Reference link: [pray_help_me_understand_super_init_args_and_kwargs](https://www.reddit.com/r/learnpython/comments/zzkp6/pray_help_me_understand_super_init_args_and_kwargs/)